src/SourceCode.Clay.OpenApi/Expressions/OasExpressionComponent.cs

```diff
@@ -49,9 +49,7 @@ internal OasExpressionComponent()
         /// <returns>The result of the operator.</returns>
         public static bool operator ==(OasExpressionComponent expression1, OasExpressionComponent expression2)
         {
-            if (ReferenceEquals(expression1, expression2)) return true;
-            if (expression1 is null || expression2 is null) return false;
-
+            if (expression1 is null) return expression2 is null;
```

Comment:
Next line: `if (expression2 is null) return false` 
(Since we now know `expression1` is not null)

Reply:
`Equals` will handle checking in `expression2 is null`.

Reply:
At the expense of a call. But OK, it's probably a wash either way.